### PR TITLE
[k8sconfiguration] Fix bug in known_hosts property and add support for other key types

### PR DESCRIPTION
**Pull request by jonathan-innis**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
k8sconfiguration

**Comment by jonathan-innis** [comment]
Missing an assignee and needing this to go under review

**Review by yungezz — APPROVED** [review]
LGTM

File: src/k8sconfiguration/azext_k8sconfiguration/custom.py [hunk 1]
```diff
@@ -229,7 +250,10 @@ def __validate_url_with_params(repository_url, ssh_private_key_set, known_hosts_
 
 
 def __validate_known_hosts(knownhost_data):
-    knownhost_str = __from_base64(knownhost_data).decode('utf-8')
+    try:
+        knownhost_str = __from_base64(knownhost_data).decode('utf-8')
+    except Exception as ex:
+        raise CLIError('Error! ssh known_hosts is not a valid utf-8 base64 encoded string') from ex
```
Comment (qwordy) [inline src/k8sconfiguration/azext_k8sconfiguration/custom.py:256]: Should CLIError be deprecated? @yungezz

Comment (jonathan-innis) [reply src/k8sconfiguration/azext_k8sconfiguration/custom.py:256]: Is there a more up to date error to throw?

Comment (fengzhou-msft) [reply src/k8sconfiguration/azext_k8sconfiguration/custom.py:256]: Please check the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md)

Comment (jonathan-innis) [reply src/k8sconfiguration/azext_k8sconfiguration/custom.py]: Updated

File: src/k8sconfiguration/HISTORY.rst [hunk 2]
```diff
@@ -3,9 +3,12 @@
 Release History
 ===============
 
-0.2.0
+0.2.1
 ++++++++++++++++++
+* Patch bug with known_hosts, add support for DSA, ECC, and ED25519 private keys
```
[the remark this answers is not in the document]
Comment (jonathan-innis) [reply src/k8sconfiguration/HISTORY.rst]: Shortened

**Comment by jonathan-innis** [comment]
@fengzhou-msft This is done and updated based on comments, please merge ASAP

File: src/k8sconfiguration/azext_k8sconfiguration/_validators.py [hunk 3]
```diff
@@ -3,9 +3,11 @@
...
-from knack.util import CLIError
+from azure.cli.core.azclierror import InvalidArgumentValueError
 
 
 def validate_configuration_type(configuration_type):
     if configuration_type.lower() != 'sourcecontrolconfiguration':
-        raise CLIError('Invalid configuration-type.  Valid value is "sourceControlConfiguration"')
+        raise InvalidArgumentValueError(
+            'Invalid configuration-type',
+            'Try specifying the valid valid value "sourceControlConfiguration"')
```
Comment (fengzhou-msft) [inline src/k8sconfiguration/azext_k8sconfiguration/_validators.py:13]:
```suggestion
            'Try specifying the valid value "sourceControlConfiguration"')
```

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**